TimeTracker/views.py

```diff
@@ -1,90 +1,210 @@
+            image_data_decoded = base64.b64decode(file_data)
+            image = Image.open(BytesIO(image_data_decoded))
+            image_io = BytesIO()
+            image.save(image_io, format='JPEG')
+            user_profile = UserProfile.objects.get(user=request.user)
+            if user_profile.avatar:
+                user_profile.avatar.delete()  # delete the old one
 
+            rand_str = ''.join(random.sample(string.ascii_letters + string.digits, 8))
+            user_profile.avatar.save(request.user.username + '_' + rand_str + '.jpg', ContentFile(image_io.getvalue()),
+                                     save=True)
+
+            return render(request, 'TimeTracker/base.html',
+                          context={'user_file': user_profile})
+        else:
+            messages.error(request, 'Invalid Image')
     else:
-        form = SignUpForm()
+        return render(request, 'TimeTracker/userInfo.html',
+                      context={'username': request.user.username})
+
+
+def report(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/report.html')
+
+
+def table(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/Group.html')
+
+
+def music(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/music.html')
+
+
+def coin(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/coin.html')
+
+
+def setting(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/setting.html')
+
+
+def badges(request):
+    if request.method == 'GET':
+        return render(request, 'TimeTracker/badges.html')
+
+
+def report(request):
+    return render(request, 'TimeTracker/report.html')
 
-    # todo need signup page
-    return render(request, 'sign up', {'form': form})
 
 def main(request):
     return render(request, 'TimeTracker/main.html')
+
+
+@login_required
+def login_main(request):
+    return render(request, 'TimeTracker/login_main.html')
+
+
+
+
+#点击submit后创建task
+def create_task(request):
+    if request.method == 'POST':
+        title = request.POST.get('title')
+        start_time = request.POST.get('startTime')
+        totalSeconds = request.POST.get('totalSeconds')
+        # 创建并保存任务对象
+        task = Task(user=request.user, title=title, startTime=start_time,totalSeconds=totalSeconds)
+        task.save()
+        return JsonResponse({'status': 'success', 'task_id': task.id})
+    return JsonResponse({'status': 'error'}, status=400)
+
+#点击delete后删除task
+def delete_task(request):
+    task_id = request.POST.get('task_id')
+    task = get_object_or_404(Task, pk=task_id, user=request.user)
+    task.delete()
+    return JsonResponse({'status': 'success'})
+
+
+
+#前端获取用户task，保证每次刷新网页都保留已创建的task
+def get_tasks(request):
+    tasks = Task.objects.filter(user=request.user).values()  # 获取当前用户的任务
```

Comment:
It may need to add more filtering. For example, completed tasks do not need to be displayed